### ISSUE TITLE
Unable to Create Labeled Video after Analyzation

### ISSUE THREAD

**Issue by alexgaudineeralexgaudineer** [DeepLabCut/DeepLabCut#2796]
### Is there an existing issue for this?

- [X] I have searched the existing issues

### Bug description

After Following steps from the Single Animal User Guide, I was able to Extract & Label Frames, train a network, and finally analyze a video. When trying to create a labeled video though, I get and error - "No unfiltered data file found in C:\dir for video and scorer".
This happens right after a successful output from the analyze function.

### Operating System

Win11

### DeepLabCut version

3.0.0rc6

### DeepLabCut mode

single animal

### Device type

gpu
NVIDIA-SMI 556.12                 Driver Version: 556.12         CUDA Version: 12.5 

### Steps To Reproduce

Followed Steps from Single animal User Guide in Project documentation,

Created a new project
Extracted Frames
Labeled Frames
Created Training Dataset
Trained Network
Analyzed Videos
Create videos

### Relevant log output

```shell
Analyzing videos with C:\conda_envs\Cat-004-Midas-2024-11-26\dlc-models-pytorch\iteration-0\Cat-004Nov26-trainset95shuffle1\train\snapshot-best-020.pt
C:\conda_envs\dlcdlc\lib\site-packages\deeplabcut\pose_estimation_pytorch\runners\base.py:80: FutureWarning: You are using `torch.load` with `weights_only=False` (the current default value), which uses the default pickle module implicitly. It is possible to construct malicious pickle data which will execute arbitrary code during unpickling (See https://github.com/pytorch/pytorch/blob/main/SECURITY.md#untrusted-models for more details). In a future release, the default value for `weights_only` will be flipped to `True`. This limits the functions that could be executed during unpickling. Arbitrary objects will no longer be allowed to be loaded via this mode unless they are explicitly allowlisted by the user via `torch.serialization.add_safe_globals`. We recommend you start setting `weights_only=True` for any use case where you don't have full control of the loaded file. Please open an issue on GitHub for any issues related to this experimental feature.
  snapshot = torch.load(snapshot_path, map_location=device)
Starting to analyze C:\conda_envs\Cat-004-Midas-2024-11-26\videos\camera-2-vid1_30fps.mp4
Video metadata:
  Overall # of frames:    614
  Duration of video [s]:  20.47
  fps:                    30.0
  resolution:             w=1920, h=1080

Running pose prediction with batch size 8
100%|████████████████████████████████████████████████████████████████████████████████| 614/614 [02:48<00:00,  3.63it/s]
The videos are analyzed. Now your research can truly start!
You can create labeled videos with 'create_labeled_video'.
If the tracking is not satisfactory for some videos, consider expanding the training set. You can use the function 'extract_outlier_frames' to extract a few representative outlier frames.

Out[3]: 'DLC_Resnet50_Cat-004Nov26shuffle1_snapshot_020'



In [5]: deeplabcut.create_labeled_video(^M
   ...:     "C:\\conda_envs\\Cat-004-Midas-2024-11-26\\config.yaml",^M
   ...:     ["C:\\conda_envs\\Cat-004-Midas-2024-11-26\\videos\\camera-2-vid1_30fps.mp4"],^M
   ...:     videotype=".mp4",^M
   ...:     trailpoints=2,^M
   ...:     draw_skeleton=True,^M
   ...:     save_frames=True^M
   ...: )
Starting to process video: C:\conda_envs\Cat-004-Midas-2024-11-26\videos\camera-2-vid1_30fps.mp4
Loading C:\conda_envs\Cat-004-Midas-2024-11-26\videos\camera-2-vid1_30fps.mp4 and data.
No unfiltered data file found in C:\conda_envs\Cat-004-Midas-2024-11-26\videos for video camera-2-vid1_30fps and scorer DLC_Resnet50_Cat-004Nov26shuffle1_snapshot_020.
```

### Anything else?

Config file as follows

```python
# Project definitions (do not edit)
Task: Cat-004
scorer: Midas
date: Nov26
multianimalproject: false
identity:


# Project path (change when moving around)
project_path: C:\conda_envs\Cat-004-Midas-2024-11-26


# Default DeepLabCut engine to use for shuffle creation (either pytorch or tensorflow)
engine: pytorch


# Annotation data set configuration (and individual video cropping parameters)
video_sets:
  C:\conda_envs\Cat-004-Midas-2024-11-26\videos\camera-1-vid1.mp4:
    crop: 746, 1094, 506, 754
  C:\conda_envs\Cat-004-Midas-2024-11-26\videos\camera-1-vid1_30fps.mp4:
    crop: 812, 1071, 526, 754
  C:\conda_envs\Cat-004-Midas-2024-11-26\videos\camera-2-vid1.mp4:
    crop: 614, 974, 375, 673
bodyparts:
- front_paw_left
- back_paw_left
- ankle_left
- knee_left
- shoulder_left

- front_paw_right
- back_paw_right
- ankle_right
- knee_right
- shoulder_right

- front_paw_left_back
- back_paw_left_back
- ankle_left_back
- knee_left_back
- hip_bottom_left
- hip_top_left

- front_paw_right_back
- back_paw_right_back
- ankle_right_back
- knee_right_back
- hip_bottom_right
- hip_top_right

- head
- ear_left
- ear_right
- nape
- spine_top
- spine_mid
- spine_bottom
- tail_base
- tail_mid
- tail_tip


# Fraction of video to start/stop when extracting frames for labeling/refinement


# Fraction of video to start/stop when extracting frames for labeling/refinement


# Fraction of video to start/stop when extracting frames for labeling/refinement


# Fraction of video to start/stop when extracting frames for labeling/refinement


# Fraction of video to start/stop when extracting frames for labeling/refinement


# Fraction of video to start/stop when extracting frames for labeling/refinement


# Fraction of video to start/stop when extracting frames for labeling/refinement
start: 0
stop: 1
numframes2pick: 20


# Plotting configuration
skeleton:
- - front_paw_left
  - back_paw_left
- - ankle_left
  - knee_left
  - shoulder_left
- - front_paw_right
  - back_paw_right
- - ankle_right
  - knee_right
  - shoulder_right
- - front_paw_left_back
  - back_paw_left_back
- - ankle_left_back
  - knee_left_back
  - hip_bottom_left
  - hip_top_left
- - front_paw_right_back
  - back_paw_right_back
- - ankle_right_back
  - knee_right_back
  - hip_bottom_right
  - hip_top_right
- - head
  - ear_left
  - ear_right
- - nape
  - spine_top
  - spine_mid
  - spine_bottom
- - tail_base
  - tail_mid
  - tail_tip

skeleton_color: black
pcutoff: 0.6
dotsize: 12
alphavalue: 0.7
colormap: rainbow


# Training,Evaluation and Analysis configuration
TrainingFraction:
- 0.95
iteration: 0
default_net_type: resnet_50
default_augmenter: default
snapshotindex: -1
detector_snapshotindex: -1
batch_size: 8
detector_batch_size: 1


# Cropping Parameters (for analysis and outlier frame detection)
cropping: false
#if cropping is true for analysis, then set the values here:
x1: 0
x2: 640
y1: 277
y2: 624


# Refinement configuration (parameters from annotation dataset configuration also relevant in this stage)
corner2move2:
- 50
- 50
move2corner: true


# Conversion tables to fine-tune SuperAnimal weights
SuperAnimalConversionTables:

``` 

Screenshot of videos folder withing project 
![image](https://github.com/user-attachments/assets/7d336795-d3a2-4c77-9793-22402468b0ad)

I tried with two different videos to make sure one wasn't corrupted or something. I tried changing the names of both and recreating, but was not working. Any help would be greatly appreciated.

### Code of Conduct

- [X] I agree to follow this project's [Code of Conduct](https://github.com/DeepLabCut/DeepLabCut/blob/master/CODE_OF_CONDUCT.md)

**Comment by n-poulsen** [comment]
Hi @alexgaudineeralexgaudineer! Thanks for reporting this, I'll open a pull request to fix the issue!

**Comment by n-poulsen** [comment]
Please update your code to get the fix:  `pip install --upgrade "git+https://github.com/DeepLabCut/DeepLabCut.git@pytorch_dlc#egg=deeplabcut"`

**Comment by swxy2160312** [comment]
> 请更新您的代码以获取修复程序：`pip install --upgrade "git+https://github.com/DeepLabCut/DeepLabCut.git@pytorch_dlc#egg=deeplabcut"`

同样的问题，运行上面命令并没有更新新的东西，问题并没有解决

**Comment by MHRosenberg** [comment]
I'm encountering what might be a similar error on linux with both the current DLC version and 2.3.5. The weird thing is that I can't find the exact error message in the repo... Even a pointer to where that relevant code lives would be helpful.

The original trackings was done with version 2.3.5 using the tensorflow engine.  

dlc.create_labeled_video('DLC_cup/MR_oT_retrained_w_all_frames-MR-2023-11-17/240522_config-nearLeft5x5.yaml', ['videos_ephys/240527-172534_a024_eM000.avi'])

With either filtered=True or False I get:

Starting to process video: /jukebox/tank/MR/maze_vids_bidirectional/videos_ephys/240527-172534_a024_eM000.avi
Loading /jukebox/tank/MR/maze_vids_bidirectional/videos_ephys/240527-172534_a024_eM000.avi and data.
No full data found in /jukebox/tank/MR/maze_vids_bidirectional/videos_ephys for video 240527-172534_a024_eM000 and scorer DLC_resnet50_MR_oT_retrained_w_all_framesNov17shuffle1_1030000.

The video, the unfiltered tracking h5, and the filtered tracking h5 all live in the same directory.

**Comment by slagathor69** [comment]
> I'm encountering what might be a similar error on linux with both the current DLC version and 2.3.5. The weird thing is that I can't find the exact error message in the repo... Even a pointer to where that relevant code lives would be helpful.
> 
> The original trackings was done with version 2.3.5 using the tensorflow engine.
> 
> dlc.create_labeled_video('DLC_cup/MR_oT_retrained_w_all_frames-MR-2023-11-17/240522_config-nearLeft5x5.yaml', ['videos_ephys/240527-172534_a024_eM000.avi'])
> 
> With either filtered=True or False I get:
> 
> Starting to process video: /jukebox/tank/MR/maze_vids_bidirectional/videos_ephys/240527-172534_a024_eM000.avi Loading /jukebox/tank/MR/maze_vids_bidirectional/videos_ephys/240527-172534_a024_eM000.avi and data. No full data found in /jukebox/tank/MR/maze_vids_bidirectional/videos_ephys for video 240527-172534_a024_eM000 and scorer DLC_resnet50_MR_oT_retrained_w_all_framesNov17shuffle1_1030000.
> 
> The video, the unfiltered tracking h5, and the filtered tracking h5 all live in the same directory.

I am having this exact same error and also used tensorflow for engine and running on Linux. 

I'm surprised you are experiencing it with 2.5 as I was guessing it was due to the latest release of deeplabcut 3.0 as I have a slightly older deeplabcut 3.0rc release on my MacBook and this error does not occur running exact same code. Also I ran this exact same code just a couple weeks ago on this same Linux computer before updating deeplabcut 3.0rc to current deeplabcut 3.0 release and had no issues. 

I believe something in the recent update broke create_labeled_videos, perhaps if only for users who used tensorflow on Linux as I have not seen anyone else describe this error which I have also been running into since the past 2 days. 

Probably best if we open a new issue seeing that this one is closed and we may be the first ones experiencing it.

@n-poulsen should we create new issue or do you want to reopen this one?

**Comment by MHRosenberg** [comment]
@slagathor69, cool, thanks for replying! I'm glad I'm not crazy (or uncommonly dumb lol). I just created a [new issue](https://github.com/DeepLabCut/DeepLabCut/issues/2825#issue-2771620005) but that issue combines several problems I'm having that might distract from the common issue we share. It might make sense to create your own issue and link my new issue or this thread as supporting info.

**Comment by n-poulsen** [comment]
@slagathor69 as you said, this seems to be an issue that only occurs with single animal `tensorflow` models and the latest release -  if you could create a new issue that would be great. I'm working on a fix that should be ready today

**Comment by n-poulsen** [comment]
@slagathor69 this issue will be fixed once #2826 is merged - I don't think it's necessary to open a new issue

**Comment by slagathor69** [comment]
> @slagathor69 this issue will be fixed once #2826 is merged - I don't think it's necessary to open a new issue

@n-poulsen Thank you! I appreciate y'all for being so responsive

**Comment by maximpavliv** [comment]
> > @slagathor69 this issue will be fixed once #2826 is merged - I don't think it's necessary to open a new issue
> 
> @n-poulsen Thank you! I appreciate y'all for being so responsive

[#2826](https://github.com/DeepLabCut/DeepLabCut/pull/2826) has been merged!

**Comment by MHRosenberg** [comment]
@slagathor69, did "pip install --upgrade "git+https://github.com/DeepLabCut/DeepLabCut.git@pytorch_dlc#egg=deeplabcut"  actually solve this error on your end? The upgrade appears to have worked but I get the identical errors.

**Comment by slagathor69** [comment]
> @slagathor69, did "pip install --upgrade "git+https://github.com/DeepLabCut/DeepLabCut.git@pytorch_dlc#egg=deeplabcut"  actually solve this error on your end? The upgrade appears to have worked but I get the identical errors. 

@MHRosenberg Nope I tried upgrading and still had same error so I manually edited the make_labeled_video.py file with the changes from the [commit](https://github.com/DeepLabCut/DeepLabCut/commit/ae612e04c14e56a71a079b12ecffae96c17acd21) and that fixed it for me. 

@n-poulsen I don't think your changes that fixed this error were pushed after all. 

I'm trying to switch to PyTorch to avoid errors like this in the future but despite what everyone says I always get better performance with TensorFlow so I'm trying out different parameters and increasing epochs but haven't had much luck yet.

**Comment by MHRosenberg** [comment]
@slagathor69, thanks for the quick reply!

**Comment by slagathor69** [comment]
> @slagathor69, wow, that's annoying. Thanks for the quick reply! After manually hacking the file you referenced, it's running on my end as well.
> 
> Have you explored SLEAP at all? I'm in the process of converting back and forth between the two. So far, everything is smoother and better maintained.
> 
> I'm becoming increasingly interested in using DLC/SLEAP exclusively for handling data labeling and just writing my own lightweight custom code to handle the model training. It seems like the need to support complex GUIs and diverse use cases will likely always make these libraries buggy and outdated relative to rapidly maturing machine vision algorithms/models.
> 
> Would you be interested in corresponding via slack or telegram (or whatever, other platform, I suppose) to compare notes and share code? It seems like you and I are working on similar use cases and hitting similar errors. If you're interested, you could reply here or email me directly at [mhr3@princeton.edu](mailto:mhr3@princeton.edu)

Sure I just sent a Slack invitation now!

**Comment by mikeumo** [comment]
I am having the same issue: cannot create a labeled video after analysis. The error is the same: 
INFO:console:Starting to process video: <name>.mp4
INFO:console:Loading <name>.mp4 and data.
INFO:console:No unfiltered data file found in <folder name> for video <name> and scorer <name>.
INFO:console:Failed to create videos from <name>.mp4.

I am using DLC 3.0.0rc6 on MacOS 15.3.1

Additionally, When I select "save Results as CSV" in Analyze videos" -- no CSV file is created, only the "full" and "meta" pickles.

Could someone help?

**Comment by slagathor69** [comment]
@n-poulsen please reopen this issue since upgrading does not fix it

@mikeumo the only way I was able to fix error was manually modifying the make_labeled_video.py file with the changes from the [commit](https://github.com/DeepLabCut/DeepLabCut/commit/ae612e04c14e56a71a079b12ecffae96c17acd21).

**Comment by mikeumo** [comment]
Well, the lines in the commit you reference are very different from the lines in my make_labeled_video.py:
the first line: "df, filepath, _, _ ..." is located at line 790, not 888. Then after line "if bp in bodyparts" at 815 things are very different of what to replace. I do not have "
frames_dict = {
                int(key.replace("frame", "")): value
                for key, value in full_data.items()
                if key.startswith("frame") and key[5:].isdigit()
            }
" after "if bp in bodyparts"

 So, I am not sure it makes sense to replace what I have in my file.

**Comment by n-poulsen** [comment]
@slagathor69 @mikeumo could you try uninstalling `deeplabcut` before reinstalling - I think that that may be what's causing the issue, if the version wasn't changed since the last install:

```bash
pip uninstall deeplabcut
pip install --upgrade "git+https://github.com/DeepLabCut/DeepLabCut.git@pytorch_dlc#egg=deeplabcut"
```

This should hopefully resolve both of these issues. If that still fails, could you try creating a fresh environment with the latest code?

**Comment by mikeumo** [comment]
I re-installed the deeplabcut as suggested and that seems to have solved the problem. 

Thank you.

**Comment by n-poulsen** [comment]
Thanks for testing this @mikeumo and sorry for providing the wrong command for updating the code! I believe running `pip install --upgrade` only updates the code if the version was updated. As in this case it was still `rc6`, uninstalling the package and reinstalling it is the only way it will work.

**Comment by mikeumo** [comment]
No problem! DLC is very useful to my research. Thank you, guys!

**Comment by mikeumo** [comment]
I am getting the same error again: "No unfiltered data file found...". I have reinstalled DLC3 3.0.0rc13 into fresh environment with python 3.12.12.
